httprequest_lego_provider/forms.py

```diff
@@ -2,8 +2,66 @@
 # See LICENSE file for licensing details.
 """Forms."""
 
+import re
+
+from django.core.exceptions import ValidationError
 from django.forms import CharField, Form
 
+FQDN_PREFIX = "_acme-challenge."
+
+
+def _is_fqdn(fqdn) -> bool:
+    """Check if the argument is a valid FQDN.
+
+    Args:
+        fqdn: the FQDN to validate.
+
+    Returns:
+        if the FQDN is valid.
+    """
+    return bool(
+        re.match(
+            (
+                r"^(?!.{255}|.{253}[^.])([a-z0-9](?:[-a-z-0-9]{0,61}[a-z0-9])?\.)+"
+                r"[a-z0-9](?:[-a-z0-9]{0,61}[a-z0-9])?[.]?$"
+            ),
+            fqdn,
+            re.IGNORECASE,
+        )
+    )
+
+
+def is_fqdn_compliant(fqdn) -> bool:
+    """Check if value consists only of a valid FQDNs prefixed by '_acme-challenge.'.
+
+    Args:
+        fqdn: the FQDN to validate.
+
+    Returns:
+        if the FQDN is valid.
+    """
+    return fqdn.startswith(FQDN_PREFIX) and _is_fqdn(fqdn.split(".", 1)[1])
+
+
+class FQDNField(CharField):
```

Comment:
I have just a small question, is there any reason why we use this and not pydantic?

Reply:
This mechanism is already provided by django, so why adding additional deps?